tsdb/chunks/chunks.go

```diff
@@ -256,19 +258,27 @@ func MergeOverlappingChunks(chks []Meta) ([]Meta, error) {
 		if c.MaxTime > nc.MaxTime {
 			nc.MaxTime = c.MaxTime
 		}
-		chk, err := MergeChunks(nc.Chunk, c.Chunk)
+		dividedChunksArray, err := MergeChunks(nc.Chunk, c.Chunk)
 		if err != nil {
 			return nil, err
 		}
-		nc.Chunk = chk
+		nc.Chunk = dividedChunksArray[0]
+		if len(dividedChunksArray) > 1 {
+			for _, chunk := range dividedChunksArray[1:] {
+				mint, maxt := getMinAndMaxTimeOfChunk(chunk)
+				newMeta := Meta{Chunk: chunk, MinTime: mint, MaxTime: maxt}
+				newChks = append(newChks, newMeta)
+				last++
+			}
+		}
 	}
-
 	return newChks, nil
 }
 
 // MergeChunks vertically merges a and b, i.e., if there is any sample
 // with same timestamp in both a and b, the sample in a is discarded.
-func MergeChunks(a, b chunkenc.Chunk) (*chunkenc.XORChunk, error) {
+// Also breaks chunks that are larger than a particular limit.
+func MergeChunks(a, b chunkenc.Chunk) ([]chunkenc.Chunk, error) {
```

Comment:
We can expect at most 2 chunks as a result. So what about having 2 chunks being returned instead of a slice? It would also simplify the logic in `MergeOverlappingChunks`.